src/test/java/com/iexec/core/task/TaskRepositoryTest.java

```diff
@@ -0,0 +1,147 @@
+package com.iexec.core.task;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.data.domain.Sort;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+
+import static com.iexec.core.task.TaskStatus.INITIALIZED;
+import static com.iexec.core.task.TaskStatus.RUNNING;
+import static com.iexec.core.task.TaskTestsUtils.getStubTask;
+
+@DataMongoTest
```

Comment:
Are we sure we want Spring tests in our unit tests?

Reply:
Those are specific tests which only create beans for entities and repositories. The beans annotated with @Components and @Services are not created.
The new tests specifically target and validate the data model which should be validated as early as possible.
There is a valid discussion on the topic and we decided to start this way and maybe to refactor later.